main/asterisk.c

```diff
@@ -3172,26 +3172,43 @@ static int ast_el_read_history(const char *filename)
 	return history(el_hist, &ev, H_LOAD, filename);
 }
 
+static char *get_default_histfile_name(void)
+{
+	const char *fn = ".asterisk_history";
+	struct passwd *pw = getpwuid(geteuid());
+	size_t bufsize = 0;
+	char *name = NULL;
+
+	if (ast_strlen_zero(pw->pw_dir)) {
+		return NULL;
+	}
+
+	/* pw_dir + '/' + fn + NULL */
+	bufsize = strlen(pw->pw_dir) + 1 + strlen(fn) + 1;
+	name = ast_malloc(bufsize);
```

Comment:
What if malloc fails?
I'm not sure I like the idea of adding an allocation rather than taking a buffer, is this mainly to deal with the truncation? I guess it's only done once per session so not really a big deal.

Reply:
Personally, my preference would be to use ast_asprintf instead of ast_malloc + NULL check + snprintf. It's not used as much but there is an ast_ API for it so that would simplify a bit.
At the very least, snprintf is unnecessary since we know the buffer is large enough, sprintf can be safely used.

Reply:
Yes, use `ast_asprintf()` instead of manually allocating.